### PR TITLE
Adjusting CMakeLists.txt for LLVM monorepo

### PR DESCRIPTION
**Pull request by AlexeySotkin**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -80,14 +80,21 @@ set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
 
 if(NOT USE_PREBUILT_LLVM)
     set(TARGET_BRANCH "ocl-open-90")
-    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+
+    if(NOT LLVM_EXTERNAL_CLANG_SOURCE_DIR)
+      set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+    elseif(EXISTS "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}/CMakeLists.txt") #monorepo
+      set(CLANG_SOURCE_DIR "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}")
+    endif()
```
Comment (AlexeySachkov) [inline CMakeLists.txt]: What if  `LLVM_EXTERNAL_CLANG_SOURCE_DIR` is set, but points to a wrong subdirectory? I think we should add `else` branch which emits error message

File: CMakeLists.txt [hunk 2]
```diff
@@ -80,14 +80,21 @@ set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
...
+    message(STATUS "CLANG_SOURCE_DIR = ${CLANG_SOURCE_DIR}")
```
Comment (AlexeySachkov) [inline CMakeLists.txt]: I think it is better to say something like "Using clang sources from /path/to/clang/sources".
My main concern is that `CLANG_SOURCE_DIR` looks like option name and someone might try to set it to override the path

File: CMakeLists.txt [hunk 3]
```diff
@@ -154,11 +161,10 @@ add_definitions( -DCOMMON_CLANG_EXPORTS )
 
 if(NOT USE_PREBUILT_LLVM)
     set(CLANG_BINARY_DIR ${LLVM_BINARY_DIR}/tools/clang/)
-    set(CLANG_SOURCE_DIR ${LLVM_MAIN_SRC_DIR}/tools/clang/)
     include_directories(
         ${CLANG_BINARY_DIR}/include # for tablegened includes
         ${CLANG_SOURCE_DIR}/include # for basic headers
-        ${SPIRV_SOURCE_DIR}/include)
+        ${SPIRV_SOURCE_DIR}/include)# for SPIRV headers
```
Comment (AlexeySachkov) [inline CMakeLists.txt]:
```suggestion
        ${SPIRV_SOURCE_DIR}/include) # for SPIRV headers
```

**Comment by AlexeySachkov** [comment]
@dvrogozh, @PawelJurek, FYI

File: CMakeLists.txt (hunk 1 above)
Comment (AlexeySotkin) [reply CMakeLists.txt]: Fixed

File: CMakeLists.txt (hunk 2 above)
Comment (AlexeySotkin) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (AlexeySotkin) [reply CMakeLists.txt]: Updated

**Pull request merged**